### PR TITLE
Correct Transaction Suppliers having default values overwritten, and add validation

### LINKED ISSUE #1303: Update Monitor to Not Overwrite Default Values of Missing Properties
**Problem** 
The monitor is not respecting the different Suppliers' default property values and is instead setting primitive default or null values when a property is missing.  For example, a `CryptoTransferTransactionSupplier` has a default `amount` of 1, but when executing through the monitor without specifying the amount in the yml, the Supplier is created with an `amount = 0`.  When switching to an object Long, the `amount = null`.

**Solution**
Correct the monitor to not overwrite these default values if properties are not set in the config.

**Additional Details**
The issue is most likely in `ConfigurableTransactionGenerator`, the `convert` method.

### PR DESCRIPTION
**Pull request by ijungmann**
**Detailed description**:
The Transaction Suppliers are not being created correctly by the monitor because they are created through the ObjectMapper, which does not make use of the Lombok builder.  This causes the class-defined default values to be overwritten by the primitive default values.  Because these two do not seem to work well together, we are removing the Builder logic from the Transaction Suppliers.  In addition, we are adding validation logic to the monitor so we can detect these issues quicker.

- Remove Lombok `@Builder` support from Transaction Suppliers and add getters/setters to all fields
-  Add validation logic to the monitor and validation annotations to the Transaction Suppliers

**Which issue(s) this PR fixes**:
Fixes #1303

**Special notes for your reviewer**:

**Checklist**
- [ ] Documentation added
- [ ] Tests updated

### CODE REVIEW

File: hedera-mirror-datagenerator/src/main/java/com/hedera/datagenerator/sdk/supplier/account/AccountUpdateTransactionSupplier.java [hunk 1]
```diff
@@ -37,30 +38,28 @@
...
 
-@Builder
-@Value
-@Log4j2
+@Data
 public class AccountUpdateTransactionSupplier implements TransactionSupplier<AccountUpdateTransaction> {
 
     private static final List<String> requiredFields = Arrays.asList("accountId");
 
     //Required
```
Comment (steven-sheehy) [inline hedera-mirror-datagenerator/src/main/java/com/hedera/datagenerator/sdk/supplier/account/AccountUpdateTransactionSupplier.java:46]: I think you can remove the Required/Optional comments now and re-sort the required fields into the optional fields. It should be self documenting what's required from the validation annotations

File: hedera-mirror-monitor/pom.xml [hunk 2]
```diff
@@ -82,6 +82,16 @@
             <artifactId>commons-math3</artifactId>
             <version>3.6.1</version>
         </dependency>
+        <dependency>
+            <groupId>javax.el</groupId>
+            <artifactId>javax.el-api</artifactId>
```
Comment (steven-sheehy) [inline hedera-mirror-monitor/pom.xml:87]: You shouldn't need this dependency, just the org.glassfish:javax.el.

File: hedera-mirror-monitor/pom.xml [hunk 3]
```diff
@@ -82,6 +82,16 @@
...
+            <version>3.0.0</version>
+        </dependency>
+        <dependency>
+            <groupId>org.glassfish</groupId>
+            <artifactId>javax.el</artifactId>
+            <version>3.0.0</version>
```
Comment (steven-sheehy) [inline hedera-mirror-monitor/pom.xml:93]: Move to property

File: hedera-mirror-datagenerator/src/main/java/com/hedera/datagenerator/sdk/supplier/account/AccountUpdateTransactionSupplier.java [hunk 4]
```diff
@@ -37,30 +38,28 @@
 import com.hedera.hashgraph.sdk.account.AccountUpdateTransaction;
 import com.hedera.hashgraph.sdk.crypto.ed25519.Ed25519PublicKey;
 
-@Builder
-@Value
-@Log4j2
+@Data
 public class AccountUpdateTransactionSupplier implements TransactionSupplier<AccountUpdateTransaction> {
 
     private static final List<String> requiredFields = Arrays.asList("accountId");
```
Comment (steven-sheehy) [inline hedera-mirror-datagenerator/src/main/java/com/hedera/datagenerator/sdk/supplier/account/AccountUpdateTransactionSupplier.java:44]: Should remove this and the manual validation checks.

File: hedera-mirror-datagenerator/src/main/java/com/hedera/datagenerator/sdk/supplier/account/AccountUpdateTransactionSupplier.java [hunk 5]
```diff
@@ -37,30 +38,28 @@
...
-    private final long maxTransactionFee = 1_000_000_000;
-    private final String proxyAccountId;
-    private final String publicKey;
+    @Min(1)
+    private long maxTransactionFee = 1_000_000_000;
 
-    @Builder.Default
-    private final boolean receiverSignatureRequired = false;
+    private String proxyAccountId;
+    private String publicKey;
```
Comment (steven-sheehy) [inline hedera-mirror-datagenerator/src/main/java/com/hedera/datagenerator/sdk/supplier/account/AccountUpdateTransactionSupplier.java:61]: nit: Would look more consistent if two newlines between all fields.

**Review by steven-sheehy — CHANGES_REQUESTED** [review]

File: hedera-mirror-datagenerator/src/main/java/com/hedera/datagenerator/sdk/supplier/account/AccountUpdateTransactionSupplier.java [hunk 6]
```diff
@@ -37,30 +38,28 @@
...
     //Required
-    private final String accountId;
+    @NotBlank
+    private String accountId;
 
     //Optional
-    @Builder.Default
-    private final Duration autoRenewPeriod = Duration.ofSeconds(8000000);
+    @NotNull
+    private Duration autoRenewPeriod = Duration.ofSeconds(8000000);
```
Comment (steven-sheehy) [inline hedera-mirror-datagenerator/src/main/java/com/hedera/datagenerator/sdk/supplier/account/AccountUpdateTransactionSupplier.java:52]: Use `@DurationMin/Max` as well for this and other duration fields

File: hedera-mirror-monitor/pom.xml (hunk 3 above)
Comment (ijungmann) [reply hedera-mirror-monitor/pom.xml:93]: Should that stay in this pom, or move to the parent pom in case we want to add this elsewhere?

Comment (steven-sheehy) [reply hedera-mirror-monitor/pom.xml:93]: Only move things to parent pom when it impacts more than one pom.xml

File: hedera-mirror-datagenerator/src/main/java/com/hedera/datagenerator/sdk/supplier/account/AccountUpdateTransactionSupplier.java (hunk 5 above)
Comment (ijungmann) [reply hedera-mirror-datagenerator/src/main/java/com/hedera/datagenerator/sdk/supplier/account/AccountUpdateTransactionSupplier.java:62]: Sure, two newlines everywhere now

File: hedera-mirror-monitor/pom.xml (hunk 2 above)
Comment (ijungmann) [reply hedera-mirror-monitor/pom.xml]: Removed

File: hedera-mirror-monitor/pom.xml (hunk 3 above)
Comment (ijungmann) [reply hedera-mirror-monitor/pom.xml]: Makes sense, moved it into a property now.

File: hedera-mirror-datagenerator/src/main/java/com/hedera/datagenerator/sdk/supplier/account/AccountUpdateTransactionSupplier.java (hunk 4 above)
Comment (ijungmann) [reply hedera-mirror-datagenerator/src/main/java/com/hedera/datagenerator/sdk/supplier/account/AccountUpdateTransactionSupplier.java]: Removed

Comment (steven-sheehy) [reply hedera-mirror-datagenerator/src/main/java/com/hedera/datagenerator/sdk/supplier/account/AccountUpdateTransactionSupplier.java]: Is the `TransactionSupplierException` in use anywhere now?

File: hedera-mirror-datagenerator/src/main/java/com/hedera/datagenerator/sdk/supplier/account/AccountUpdateTransactionSupplier.java (hunk 1 above)
Comment (ijungmann) [reply hedera-mirror-datagenerator/src/main/java/com/hedera/datagenerator/sdk/supplier/account/AccountUpdateTransactionSupplier.java]: Removed and re-sorted

File: hedera-mirror-datagenerator/src/main/java/com/hedera/datagenerator/sdk/supplier/account/AccountUpdateTransactionSupplier.java (hunk 6 above)
Comment (ijungmann) [reply hedera-mirror-datagenerator/src/main/java/com/hedera/datagenerator/sdk/supplier/account/AccountUpdateTransactionSupplier.java:45]: Added `@DurationMin` for all Durations, as well as `@Future` for all Instants

File: hedera-mirror-datagenerator/src/main/java/com/hedera/datagenerator/sdk/supplier/account/AccountUpdateTransactionSupplier.java (hunk 4 above)
Comment (ijungmann) [reply hedera-mirror-datagenerator/src/main/java/com/hedera/datagenerator/sdk/supplier/account/AccountUpdateTransactionSupplier.java]: Nope, just removed it.

**Review by Nana-EC — DISMISSED** [review]
LGTM

File: hedera-mirror-datagenerator/src/main/java/com/hedera/datagenerator/sdk/supplier/account/CryptoTransferTransactionSupplier.java [hunk 7]
```diff
@@ -20,45 +20,37 @@
...
+    @NotBlank
+    private String recipientAccountId;
 
-    @Builder.Default
-    private final long maxTransactionFee = 1_000_000;
+    @NotBlank
+    private String senderAccountId;
 
-    private final String tokenId;
+    private String tokenId;
```
Comment (steven-sheehy) [inline hedera-mirror-datagenerator/src/main/java/com/hedera/datagenerator/sdk/supplier/account/CryptoTransferTransactionSupplier.java:49]: Later, not part of this PR, we can consider creating a custom annotation and validator for these ID fields. Either that or use a wrapper object to validate.

File: hedera-mirror-datagenerator/src/main/java/com/hedera/datagenerator/sdk/supplier/consensus/ConsensusUpdateTopicTransactionSupplier.java [hunk 8]
```diff
@@ -23,49 +23,41 @@
...
+    private String adminKey;
 
-    //Required
-    private final String topicId;
+    private String autoRenewAccountId;
 
-    //Optional
-    private final String adminKey;
-    private final String autoRenewAccountId;
+    @DurationMin(seconds = 1)
```
Comment (steven-sheehy) [inline hedera-mirror-datagenerator/src/main/java/com/hedera/datagenerator/sdk/supplier/consensus/ConsensusUpdateTopicTransactionSupplier.java:46]: You need to keep `@NotNull` as `@DurationMin` still allows null.

File: hedera-mirror-datagenerator/src/main/java/com/hedera/datagenerator/sdk/supplier/consensus/ConsensusUpdateTopicTransactionSupplier.java [hunk 9]
```diff
@@ -23,49 +23,41 @@
...
+    private Duration autoRenewPeriod = Duration.ofSeconds(8000000);
 
-    @Builder.Default
-    private final Duration autoRenewPeriod = Duration.ofSeconds(8000000);
+    @Future
```
Comment (steven-sheehy) [inline hedera-mirror-datagenerator/src/main/java/com/hedera/datagenerator/sdk/supplier/consensus/ConsensusUpdateTopicTransactionSupplier.java:49]: Same for `@Future`

**Review by steven-sheehy — CHANGES_REQUESTED** [review]

File: hedera-mirror-datagenerator/src/main/java/com/hedera/datagenerator/sdk/supplier/account/CryptoTransferTransactionSupplier.java (hunk 7 above)
Comment (ijungmann) [reply hedera-mirror-datagenerator/src/main/java/com/hedera/datagenerator/sdk/supplier/account/CryptoTransferTransactionSupplier.java:49]: I actually started looking into that and decided to hold off for now, but I like the idea.

File: hedera-mirror-datagenerator/src/main/java/com/hedera/datagenerator/sdk/supplier/consensus/ConsensusUpdateTopicTransactionSupplier.java (hunk 8 above)
Comment (ijungmann) [reply hedera-mirror-datagenerator/src/main/java/com/hedera/datagenerator/sdk/supplier/consensus/ConsensusUpdateTopicTransactionSupplier.java:48]: Ah, did not see that, corrected now.

File: hedera-mirror-datagenerator/src/main/java/com/hedera/datagenerator/sdk/supplier/consensus/ConsensusUpdateTopicTransactionSupplier.java (hunk 9 above)
Comment (ijungmann) [reply hedera-mirror-datagenerator/src/main/java/com/hedera/datagenerator/sdk/supplier/consensus/ConsensusUpdateTopicTransactionSupplier.java:52]: Corrected

**Review by steven-sheehy — APPROVED** [review]
LGTM

**Pull request merged**